akka-http-core/src/main/scala/akka/http/impl/model/parser/SimpleHeaders.scala

```diff
@@ -226,6 +230,14 @@ private[parser] trait SimpleHeaders { this: Parser with CommonRules with CommonA
     rule { oneOrMore(addr).separatedBy(listSep) ~ EOI ~> (`X-Forwarded-For`(_)) }
   }
 
+  def `x-forwarded-host` = rule {
+    runSubParser(newUriParser(_).`relaxedHost-pushed`) ~ EOI ~> (`X-Forwarded-Host`(_))
```

Comment:
Hmm, calling the UriParser is a bit weird. Is there precedence for doing it that way?

Reply:
I'll take a look at the `Host` rule and try to re-use it.

Reply:
Let's see if this is what you meant.